src/main/java/com/salesforce/dataloader/security/EncryptionAesUtil.java

```diff
@@ -0,0 +1,255 @@
+
+    /**
+     * detect the operating system from the os.name System property and cache the result
+     *
+     * @returns - the operating system detected
+     */
+    public static OSType getOperatingSystemType() {
+        if (detectedOS == null) {
+            String OS = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
+            if ((OS.indexOf("mac") >= 0) || (OS.indexOf("darwin") >= 0)) {
+                detectedOS = OSType.MacOS;
+            } else if (OS.indexOf("win") >= 0) {
+                detectedOS = OSType.Windows;
+            } else if (OS.indexOf("nux") >= 0) {
+                detectedOS = OSType.Linux;
+            } else {
+                detectedOS = OSType.Other;
+            }
+        }
+        return detectedOS;
+    }
+
+    // Support single text encryption and decryption
+
+    static public Cipher cipher;
+    static public byte[] cipherKey;
+
+    // 16 bytes was used for 128 bit AES encryption
+    public static final int ENCRYPTION_KEY_LENGTH_IN_BYTES = 16;
+    public static final String DEFAULT_KEYFILE_NAME = "dataLoader.key";
+
+    static {
+        try {
+            cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+            detectedOS = getOperatingSystemType();
+
+        } catch (Exception e) {
+            LOGGER.error("Fail to initialize encryption: " + e.getMessage());
+            throw new RuntimeException("Fail to initialize encryption:  ", e);
+        }
+    }
+
+
+    private byte[] extractIvBytes(byte[] cipheredText) {
+        byte[] iv = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        System.arraycopy(cipheredText, 0, iv, 0, ENCRYPTION_KEY_LENGTH_IN_BYTES);
+        return iv;
+    }
+
+    private byte[] extractCipherContent(byte[] cipheredText) {
+        byte[] cipheredPassword = new byte[cipheredText.length - ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        System.arraycopy(cipheredText, ENCRYPTION_KEY_LENGTH_IN_BYTES, cipheredPassword, 0, cipheredText.length - ENCRYPTION_KEY_LENGTH_IN_BYTES);
+        return cipheredPassword;
+
+    }
+
+    private byte[] concatenateByteArray(byte[] a, byte[] b) {
+
+        byte[] c = new byte[a.length + b.length];
+        System.arraycopy(a, 0, c, 0, a.length);
+        System.arraycopy(b, 0, c, a.length, b.length);
+        return c;
+    }
+
+    public byte[] generateEncryptionKey() {
+        // The secure random uses threadId and tick to make sure it is not repeatable
+        // If a call to setSeed had not occurred previously,
+        // the first call to this method forces this SecureRandom object to seed itself.
+        // This self-seeding will not occur if setSeed was previously called.
+        SecureRandom random = new SecureRandom();
+        byte bytes[] = new byte[ENCRYPTION_KEY_LENGTH_IN_BYTES];
+        random.nextBytes(bytes);
+        return bytes;
+    }
+
+    public String createUserProfileKeyName() {
+        String path = Paths.get(System.getProperty("user.home"), ".dataloader").toString();
```

Comment:
Convention on Windows would put this under AppData (probably under the Local subdirectory, though as of writing this the config dirs are stored in Roaming) rather than directly under the profile, but this works.

Just bear in mind that if the config file roams but the keyfile doesn't (and IMO it shouldn't) then the config files will be useless on other machines until they are updated to contain passwords encrypted using keyfiles local to those machines.